CHANGELOG.md

```diff
@@ -4,6 +4,151 @@ The format is based on [Keep a Changelog].
 
 [Keep a Changelog]: http://keepachangelog.com/en/1.0.0/
 
+## [v0.25.0] - 2025-04-24
+
+A new breaking release which has been in the making for a while and the biggest change is that the
+`RpcServiceT trait` has been changed to support both the client and server side:
+
+```rust
+pub trait RpcServiceT {
+	/// Response type for `RpcServiceT::call`.
+	type MethodResponse;
+	/// Response type for `RpcServiceT::notification`.
+	type NotificationResponse;
+	/// Response type for `RpcServiceT::batch`.
+	type BatchResponse;
+
+	/// Processes a single JSON-RPC call, which may be a subscription or regular call.
+	fn call<'a>(&self, request: Request<'a>) -> impl Future<Output = Self::MethodResponse> + Send + 'a;
+
+	/// Processes multiple JSON-RPC calls at once, similar to `RpcServiceT::call`.
+	///
+	/// This method wraps `RpcServiceT::call` and `RpcServiceT::notification`,
+	/// but the root RPC service does not inherently recognize custom implementations
+	/// of these methods.
+	///
+	/// As a result, if you have custom logic for individual calls or notifications,
+	/// you must duplicate that implementation in this method or no middleware will be applied
+	/// for calls inside the batch.
+	fn batch<'a>(&self, requests: Batch<'a>) -> impl Future<Output = Self::BatchResponse> + Send + 'a;
+
+	/// Similar to `RpcServiceT::call` but processes a JSON-RPC notification.
+	fn notification<'a>(&self, n: Notification<'a>) -> impl Future<Output = Self::NotificationResponse> + Send + 'a;
+}
+```
+
+The reason for this change is to make it work for the client-side as well as make it easier to
+with return `impl Future` instead require an associated type for the `Future` which in many cases
+requires BoxFuture or ResponseFuture which improves ergonomics.
```

Comment:
Nit:

```suggestion
The reason for this change is to make it work for the client-side as well as make it easier to
implement performantly by relying on `impl Future` instead of requiring an associated type for the `Future` (which in many cases requires boxing).
```

Maybe?